src/app/hooks/use-search-parameters-validation.ts

```diff
@@ -38,9 +46,52 @@ export const useSearchParametersValidation = () => {
 		}
 	};
 
+	const validateSign = async ({ parameters, profile, network }: ValidateParameters) => {
+		const message = parameters.get("message");
+		const address = parameters.get("address");
+
+		if (!message) {
+			throw new Error(t("TRANSACTION.VALIDATION.MESSAGE_MISSING"));
+		}
+
+		if (address) {
+			const coin: Coins.Coin = profile.coins().set(network.coin(), network.id());
+
+			await coin.__construct();
+
+			const isValid = await coin.address().validate(address);
+
+			if (!isValid) {
+				throw new Error(t("TRANSACTION.VALIDATION.NETWORK_MISMATCH"));
+			}
+		}
+	};
+
 	const methods = {
+		sign: {
+			path: ({ profile, parameters, network }: PathProperties) => {
+				const address = parameters.get("address");
+
+				if (address) {
+					const wallet = profile.wallets().findByAddressWithNetwork(address, network.id())!;
+
+					return `${generatePath(ProfilePaths.SignMessageWallet, {
+						profileId: profile.id(),
+						walletId: wallet.id(),
+					})}?${parameters.toString()}`;
+				}
+
+				return `${generatePath(ProfilePaths.SignMessage, {
+					profileId: profile.id(),
+				})}?${parameters.toString()}`;
+			},
```

Comment:
```suggestion
			path: ({ profile, parameters }: PathProperties) =>
				`${generatePath(ProfilePaths.SignMessage, {
					profileId: profile.id(),
				})}?${parameters.toString()}`,
```

When deeplinking we always use the paths without wallet context.